### DISCUSSION TITLE
Existem vagas para UI Developers?

### DISCUSSION THREAD

**Discussion by rubenmarcus** [frontendbr/forum#606] [Discussão]
Estive vendo aqui alguns tópicos, e definições do nosso mundo front-end, e vi que me encaixo mais como UI Developer do que Front-end em si.

Lendo do [Stack Overflow](https://stackoverflow.com/questions/9700286/whats-the-difference-between-a-frontend-vs-ui-developer) 

> Front-End and UI Developers have the same skill set, one that focuses on either will be able to do the other one to a degree. The only difference is the focus of their skills. One might be more visual heavy (which implies using more CSS and HTML, and some design skills), and the other more logic heavy (JavaScript). Most of the time the same people can do the same work. Both are also often expected to have some form of design exposure as well if using 'developer' term. 'Engineers' are usually more logic focused and required to partain to back-end architecture principles, usually with no design. 

Eu gosto de JS, lógica e etc, e tenho certa facilidade, mas não como tenho para HTML e CSS, que é o que eu mais gosto de fazer. Recentemente venho trabalhando com React e Node, e tenho gostado bastante, mas mesmo assim o que eu mais gosto é de pensar na arquitetura da tela, na experiência do usuário, na performance do CSS, a arquitetura, a semântica do html.

Mas estava vendo aqui em questão de mercado, no Brasil achei 2 vagas apenas para UI Developer, enquanto que front-end você acha mais de 50. Lá fora muda um pouco a situação, você acha 20 vagas de UI e 10x isso de Front.

Vocês acham que existe futuro para desenvolvimento focado em UI/UX , ou obrigatoriamente, todos devemos dominar JS e pelo menos um pouco de desenvolvimento back-end, para num futuro próximo não estarmos fora do mercado?

**Comment by felipefialho** [comment]
Sua dúvida vai bem de encontro com o [último artigo que escrevi](https://medium.com/@lfeh/o-inicio-o-fim-e-o-meio-do-desenvolvimento-front-end-dfc5a123b90f) e resumidamente, penso que a tendência é cada vez mais nos tornarmos software engineer e ter skills amplas de engenharia.

Mas claro, normal que toda equipe tenha perfis multidisciplinares e alguns sejam um pouco mais inclinados para a parte visual, mas da forma que você disso, acho que vai fazer cada vez menos sentido no mercado apesar de também curtir muito UI.

**Comment by bernardodiasc** [comment]
Acho que a demanda é maior para tarefas mais difíceis. Não que UI/UX não seja tão importante quanto, mas atividades de engenharia e ciências requerem muito mais capacitação e por isso a oferta de profissionais é naturalmente menor, havendo então uma demanda aparente maior.

**Comment by theandersonn** [comment]
@nandomoreirame :eyes: :point_up_2:

**Comment by lucianobarauna** [comment]
@LFeh bom artigo.
@rubenmarcus Eu sou front-end e me formei em design justamente para pegar mais skills.
Já estou a 6 anos na área e comecei só no html e cs. Já tem um ano eu acho que comecei a focar em JS por conta da necessidade de estar sempre precisando dele para interações e etc.. e sem falar também com essa demanda  no mercado por javascript. Hoje eu penso assim quanto mais JS você souber mais chance de vagas você terá e mais fácil sua vida vai ficar.
PS: JS é lindo rs.

**Comment by abnersajr** [comment]
Cada vez mais o nosso mundo de front-end está demandando mais conhecimento além do tradicional HTML, CSS e JS. Quando falo JS é o "básico". Aquele maroto jQuery ou um _JSzin_ de leve só para alguma feature. Como o @LFeh comentou, cada vez mais Skills de engenharia são requeridas. Obviamente existe ainda empresas que peguem alguém apenas para a UI, mas até para isso você precisa de skills mais avançadas em JS. Afinal você irá precisar implementar um módulo, com componentes baseado em algum framework e etc.
Mas se eu pude-se te dar uma dica para não ficar para trás é estude o JS um pouco além do requerido normalmente a um UI Developer. Eu mesmo ao mudar de área tive que me esforçar um pouco mais, afinal o mercado está pedindo mais e eu queria dar um passo além, aquela saída da zona de conforto.

**Comment by kvnol** [comment]
http://vagas.design/ :heart:

**Comment by rubenmarcus** [comment]
@kvnol  ta dando 404 aqui mano

**Comment by kvnol** [comment]
@rubenmarcus sem o https rolou :smile:
